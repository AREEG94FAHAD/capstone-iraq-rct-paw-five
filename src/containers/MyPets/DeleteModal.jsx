import React,{useState,useEffect} from 'react'
import { Modal, Button } from 'react-bootstrap'
import { PropTypes } from 'prop-types'
import { useTranslation } from 'react-i18next'
import { useDispatch,useSelector } from 'react-redux'
import { deleteMyPett } from '../../redux/actions/actions'


export default function DeleteModal({
  show,
  handleClose,
  petName,
  petId,
  petImge,
}) {

  const globaleLang = useSelector(state => state.langReducer)

  const { t } = useTranslation()
  const dispatch = useDispatch()
  const deletPet = (e, id) => {
    dispatch(deleteMyPett(id))
    handleClose()
    return false
  }


  const [dirProperties, setDir] = useState({
    dir: 'ltr',
    textDir: 'text-left',
  })

  const handelDir = () => {
    if (
      globaleLang !== 'en' &&
      globaleLang !== null
    ) {
      const newDir = 'rtl'
      const newtextDir = 'text-right'
      const newdirProperties = { ...dirProperties }
      newdirProperties.dir = newDir
      newdirProperties.textDir = newtextDir
      setDir(newdirProperties)
    } else {
      const newDir = 'ltr'
      const newtextDir = 'text-left'
      const newdirProperties = { ...dirProperties }
      newdirProperties.dir = newDir
      newdirProperties.textDir = newtextDir
      setDir(newdirProperties)
    }
  }
  useEffect(() => {
    handelDir()
  }, [globaleLang])
  return (
    <div  >
      <Modal show={show} onHide={handleClose} dir={dirProperties.dir}   >
        <div>
          <Modal.Header closeButton />
          <Modal.Body className={`text-danger ${dirProperties.textDir}`}>
            <img src={petImge} alt="" height="40px" width="40px"  />
            <h7 className='p-1 text-center'>{t('mypets.warningmessage')}</h7> {petName}
          </Modal.Body>
          <Modal.Footer>
            <Button
              className="submit-button border-0"
              onClick={() => deletPet(petId)}
            >
             {t('mypets.yes')}
            </Button>
            <Button className="submit-button border-0" onClick={handleClose}>
            {t('mypets.no')}
            </Button>
          </Modal.Footer>
        </div>
      </Modal>
    </div>
  )
}

DeleteModal.propTypes = {
  show: PropTypes.bool,
  handleClose: PropTypes.func,
  petName: PropTypes.string,
  petId: PropTypes.string,
  petImge: PropTypes.string,
}

DeleteModal.defaultProps = {
  show: false,
  handleClose: false,
  petName: '',
  petId: '',
  petImge: '',
}
