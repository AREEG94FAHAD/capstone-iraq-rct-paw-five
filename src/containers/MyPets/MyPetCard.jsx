import React from 'react'

export default function MyPetCard() {
    return (
        <div>
            
        </div>
    )
}
