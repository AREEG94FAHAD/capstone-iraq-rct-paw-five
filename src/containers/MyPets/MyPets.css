.paginationBttns {
  width: 80%;
  height: 40px;
  list-style: none;
  display: flex;
  justify-content: center;
  padding: inherit;
  color: rgb(134, 134, 132);
}

.paginationBttns a {
  padding: 10px;
  margin: 8px;
  border-radius: 5px;
  border: 1px solid #b3a5a586;
  color: rgb(247, 189, 113);
  cursor: pointer;
}

.paginationBttns a:hover {
  color: white !important;
  background-color: #f37a69;
}

.paginationActive a {
  color: white !important;
  background-color: #f37a69;
  outline: none !important;
}

.paginationDisabled a {
  color: rgb(247, 189, 113);
  background-color: rgb(255, 255, 254);
}
ul .paginationBttns {
  padding: inherit !important;
}

.modal-backdrop.fade {
  opacity: 0;
  filter: alpha(opacity=0);
}
.modal-backdrop.in {
  opacity: 0.5;
  filter: alpha(opacity=50);
}

.modal-backdrop.fade {
  opacity: 0;
  filter: alpha(opacity=0);
}
.modal-backdrop.fade.in {
  opacity: 0.5;
  filter: alpha(opacity=50);
}

.mypet{
  background-color: #ffc6a975;
  background-image: linear-gradient(315deg, #f3f0f038 0%, #f1c18a02 74%);
}

.bg-select{
  background-color: azure !important;
  border-radius: 20px !important;
}
.left-raduis{
  border-top-left-radius: 20px !important;
  border-bottom-left-radius:20px ;
}

.right-raduis{
  border-bottom-right-radius: 20px !important;
  border-top-right-radius: 20px !important;
}