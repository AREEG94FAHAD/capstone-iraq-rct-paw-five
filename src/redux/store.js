import { createStore, applyMiddleware } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import logger from 'redux-logger'
import combineReducers from './reducers/AllReducers'

const store = createStore(
  combineReducers,
  composeWithDevTools(applyMiddleware(logger))
)

export default store
