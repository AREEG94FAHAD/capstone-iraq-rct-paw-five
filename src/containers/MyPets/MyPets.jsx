import React, { useEffect, useState } from 'react'
import ReactPaginate from 'react-paginate'
import './MyPets.css'
import {
  Col,
  Row,
  Form,
  InputGroup,
  FormControl,
  Container,
} from 'react-bootstrap'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faSearch, faTrash } from '@fortawesome/free-solid-svg-icons'
import { useSelector, useDispatch } from 'react-redux'
import DeleteModal from './DeleteModal'
import 'bootstrap/dist/css/bootstrap.min.css'
import { fetchMyPets, UpdateMyPett } from '../../redux/actions/actions'

export default function FetchMyPets() {
  const [pageNumber, setPageNumber] = useState(0)

  const MyPets = useSelector(state => state.myPets)
  const usersPerPage = Math.ceil(MyPets.length / 3)
  const pagesVisited = pageNumber * usersPerPage
  const pageCount = Math.ceil(MyPets.length / usersPerPage)

  const [show, setShow] = useState(false)

  const [petId, setPetId] = useState('')
  const [petName, setPetName] = useState('')
  const [petImg, setPetImg] = useState('')

  const handleClose = () => setShow(false)

  const handleShow = (e, id, name,img) => {
    setPetId(id)
    setPetName(name)
    setPetImg(img)
    setShow(true)
  }

  const changePage = ({ selected }) => {
    setPageNumber(selected)
  }

  const dispatch = useDispatch()
  const [name, setname] = useState('')
  const [sort, setsort] = useState('petName')

  const selectedType = e => {
    setname('')
    setsort(e.target.value)
  }

  const adoptTogle = (e, id, adopted) => {
    dispatch(UpdateMyPett(id, adopted))
  }

  const changeName = e => {
    setsort('petName')
    setname(e.target.value)
  }

  useEffect(() => {
    dispatch(fetchMyPets(name, sort))
  }, [name, sort])

  return (
    <div>
      <Container className=" slid p-5" fluid>
        <h2 className="text-center text-white p-3">Welcome Paw-five</h2>
        <Row>
          <Col className="col-7 d-flex justify-content-center">
            <InputGroup className="w-75 text-center">
              <FormControl
                id="inlineFormInputGroupUsername"
                placeholder="Pet Name"
                className="text-danger text-size"
                onChange={changeName}
                value={name}
              />
              <InputGroup.Prepend>
                <InputGroup.Text>
                  <FontAwesomeIcon icon={faSearch} className="text-size" />{' '}
                </InputGroup.Text>
              </InputGroup.Prepend>
            </InputGroup>
          </Col>
          <Col className="col-5">
            <Form.Control
              as="select"
              className="mr-sm-2 text-size bg-white w-75"
              id="inlineFormCustomSelect"
              custom
              value={sort}
              onChange={selectedType}
            >
              <option value="petName">Name</option>
              <option value="age">Age</option>
            </Form.Control>
          </Col>
        </Row>
      </Container>
      <Container className="img3  ">
        {MyPets
          ? MyPets.slice(pagesVisited, pagesVisited + usersPerPage).map(pet => (
              <div
                className="card text-center m-lg-5 m-md-3 m-sm-2 m-2   "
                key={pet.id}
              >
                <h5 className="card-header text-left">{pet.petName}</h5>
                <Row className="d-flex align-items-center">
                  <Col lg={3} md={6} className="d-flex justify-content-center">
                    {' '}
                    <img
                      className="card-img-top p-2 img-fluid"
                      style={{
                        height: 'auto',
                        minWidth: '200px',
                      }}
                      src={pet.file}
                      alt=""
                    />
                  </Col>

                  <Col>
                    <div className="card-body justify-content-center">
                      <h5 className="card-title text-center">
                        Special title treatment
                      </h5>
                      <p className="card-text text-center">
                        {pet.story.substr(0, 100)}
                      </p>
                      <div className="d-flex align-items-center float-right">
                        <Form.Check
                          type="switch"
                          id={pet.id}
                          label="Adopted"
                          className="align-items-center"
                          onChange={e => adoptTogle(e, pet.id, !pet.adopted)}
                          checked={pet.adopted}
                        />
                        <button
                          onClick={e => handleShow(e, pet.id, pet.petName, pet.file)}
                          key={pet.id}
                          className="btn btn-danger  m-2 text-danger bg-transparent border-0 p-3"
                          type="button"
                        >
                          <FontAwesomeIcon icon={faTrash} />
                        </button>
                      </div>
                    </div>
                  </Col>
                </Row>
              </div>
            ))
          : ''}
        <DeleteModal
          show={show}
          petId={petId}
          handleClose={handleClose}
          petName={petName}
          petImge={petImg}
        />
      </Container>
      <Container align="left" className="d-flex justify-content-center">
        <ReactPaginate
          previousLabel="<"
          nextLabel=">"
          pageCount={pageCount}
          onPageChange={changePage}
          containerClassName="paginationBttns"
          previousLinkClassName="previousBttn"
          nextLinkClassName="nextBttn"
          disabledClassName="paginationDisabled"
          activeClassName="paginationActive"
        />
      </Container>
    </div>
  )
}
