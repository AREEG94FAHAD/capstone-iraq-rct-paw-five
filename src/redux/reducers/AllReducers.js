import { combineReducers } from 'redux'
import petReducer, {
  langReducer,
  myPets,
  deleteMyPet,
  updateMyPet,
} from './petReducer'
import user from './user'

export default combineReducers({
  langReducer,
  pets: petReducer,
  user,
  myPets,
  deleteMyPet,
  updateMyPet,
})
