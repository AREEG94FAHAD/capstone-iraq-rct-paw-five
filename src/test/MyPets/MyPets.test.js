import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import { BrowserRouter as Router } from 'react-router-dom'
import MyPets from '../../containers/MyPets/MyPets'
import AllReducers from '../../redux/reducers/AllReducers'
const store = createStore(AllReducers, applyMiddleware(thunk))

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <MyPets store={store} />
        </Router>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
