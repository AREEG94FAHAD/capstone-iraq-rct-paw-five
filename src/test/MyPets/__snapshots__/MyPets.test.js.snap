// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className=" slid p-5 container-fluid"
  >
    <h2
      className="text-center text-white p-3"
    >
      mypets.welcome
       Paw-five
    </h2>
    <div
      className="row"
    >
      <div
        className="col-7 d-flex justify-content-center col"
      >
        <div
          className="w-75 text-center input-group"
        >
          <input
            className="text-danger text-size left-raduis form-control"
            id="inlineFormInputGroupUsername"
            onChange={[Function]}
            placeholder="mypets.petname"
            value=""
          />
          <div
            className="input-group-prepend"
          >
            <span
              className="right-raduis input-group-text"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-search fa-w-16 text-size"
                data-icon="search"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
               
            </span>
          </div>
        </div>
      </div>
      <div
        className="col-5  col"
      >
        <select
          className="mr-sm-2 text-size bg-select w-75 custom-select"
          id="inlineFormCustomSelect"
          onChange={[Function]}
          value="petName"
        >
          <option
            value="petName"
          >
            mypets.name
          </option>
          <option
            value="age"
          >
            mypets.age
          </option>
          <option
            value="city"
          >
            mypets.city
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="img3   container"
  >
    <div />
  </div>
  <div
    align="left"
    className="d-flex justify-content-center mt-4 container"
  >
    <ul
      className="paginationBttns"
    >
      <li
        className="previous paginationDisabled"
      >
        <a
          aria-disabled="true"
          aria-label="Previous page"
          className="previousBttn"
          onClick={[Function]}
          onKeyPress={[Function]}
          rel="prev"
          role="button"
          tabIndex="0"
        >
          &lt;
        </a>
      </li>
      <li
        className="next"
      >
        <a
          aria-disabled="false"
          aria-label="Next page"
          className="nextBttn"
          onClick={[Function]}
          onKeyPress={[Function]}
          rel="next"
          role="button"
          tabIndex="0"
        >
          &gt;
        </a>
      </li>
    </ul>
  </div>
</div>
`;
